### ISSUE TITLE
Spike: Is it worth doing TDD right from the start of Essentials?

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#22]
### Expected result

The title says it all. E.g. is there even a better testing framework than Xunit? What mocking library would we use? Would it take more than 1 day of study to start mocking?

### Additional context

Add any other context or guidelines here.

### Resources

* https://xunit.net/
* https://github.com/Moq/moq4
* https://github.com/FakeItEasy/FakeItEasy

### Issue expands upon

This issue expands upon another issue(s): #`issue number`

### Blockers

This issue is blocked by the following issue(s): #`issue number`
